### DISCUSSION TITLE
rate limit tweepy paginator search_all_tweets

### DISCUSSION THREAD

**Discussion by joostjansen** [tweepy/tweepy#1907] [Questions] [answered]
Im not sure why I am getting rate limited so quickly using:

```
mentions = []


for tweet in tweepy.Paginator(client.search_all_tweets, query= "to:######## lang:nl -is:retweet",
                            start_time = "2022-01-01T00:00:00Z", end_time = "2022-05-31T00:00:00Z",
                            max_results=500).flatten(limit=10000):
  mention = tweet.text
  mentions.append(mention)
```

I suppose I could put time.sleep(1) after these lines, but then it would mean I could only process one tweet every second, whereas with a regular client.search_all_tweets I would get 500 tweets per request.

Is there anything I'm missing here? How can I process more than one tweet a second using tweepy.Paginator?

BTW: I have academic access and know the rate limit documentation.

**Comment by Harmon758** [comment, marked as answer]
See [the FAQ section in the documentation about this](https://tweepy.readthedocs.io/en/v4.10.0/faq.html#why-am-i-getting-rate-limited-so-quickly-when-using-client-search-all-tweets-with-paginator).

This is also a duplicate of #1688 and #1871.
